src/wallet/wallet.cpp

```diff
@@ -1861,23 +1865,40 @@ std::pair<mapSaplingNoteData_t, SaplingIncomingViewingKeyMap> CWallet::FindMySap
     // Protocol Spec: 4.19 Block Chain Scanning (Sapling)
```

Comment:
The spec's algorithm does not take into account the case where we only have an incoming viewing key (see https://github.com/zcash/zips/issues/194#issuecomment-511190850 ). Bear with me while I update the algorithm in the spec, and then I'll review this section again.